### PR TITLE
TEST: implement EXPECT_FILE and EXPECT_JSON

### PR DESCRIPTION
**Pull request by luigirizzo**
Two variants of the EXPECT directive to simplify writing tests

EXPECT_FILE compares the output with the content of a file EXPECT_JSON compares the output with the content of a json file.
  It also adds "-q -f json" arguments if the program is specified
  with the PROG directive.

In both cases, the output up to __BPFTRACE_NOTIFY_PROBES_ATTACHED and leading and trailing empty lines are removed to simplify the comparison.

Two new sample test cases are added in tests/runtime/call: hist_2_values uses EXPECT_FILE, hist_10g_new uses EXPECT_JSON

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Review by danobi — COMMENTED** [review]
Pls document new directives in https://github.com/iovisor/bpftrace/blob/master/tests/README.md#runtime-test-directives .

Otherwise lgtm

File: tests/runtime/call [hunk 1]
```diff
@@ -199,6 +199,16 @@ EXPECT @bytes: *\n[\[(].*
 AFTER ./testprogs/syscall read
 TIMEOUT 5
 
+NAME hist_2_values
+PROG BEGIN { @=hist(3);@=hist(20);exit();}
+EXPECT_FILE runtime/outputs/hist_2_values.txt
+TIMEOUT 1
+
+NAME hist_10g_new
```
[the remark this answers is not in the document]
Comment (luigirizzo) [reply tests/runtime/call]: done

**Comment by luigirizzo** [comment]
> Pls document new directives in https://github.com/iovisor/bpftrace/blob/master/tests/README.md#runtime-test-directives .
> 
> Otherwise lgtm

done

**Review by danobi — APPROVED** [review]
thanks!

**Pull request merged**